platform/linux-generic/include/odp_schedule_if.h

```diff
@@ -15,6 +15,49 @@ extern "C" {
 #include <odp_queue_internal.h>
 #include <odp/api/schedule.h>
 
+/* Internal abstract queue type */
+typedef void* queue_t;
+
+/* Queue API functions */
+typedef struct {
+	odp_queue_t (*queue_create)(const char *name,
+				    const odp_queue_param_t *param);
+	int (*queue_destroy)(odp_queue_t queue);
+	odp_queue_t (*queue_lookup)(const char *name);
+	int (*queue_capability)(odp_queue_capability_t *capa);
+	int (*queue_context_set)(odp_queue_t queue, void *context,
+				     uint32_t len);
+	void *(*queue_context)(odp_queue_t queue);
+	int (*queue_enq)(odp_queue_t queue, odp_event_t ev);
+	int (*queue_enq_multi)(odp_queue_t queue, const odp_event_t events[],
+			       int num);
+	odp_event_t (*queue_deq)(odp_queue_t queue);
+	int (*queue_deq_multi)(odp_queue_t queue, odp_event_t events[],
+				   int num);
+	odp_queue_type_t (*queue_type)(odp_queue_t queue);
+	odp_schedule_sync_t (*queue_sched_type)(odp_queue_t queue);
+	odp_schedule_prio_t (*queue_sched_prio)(odp_queue_t queue);
+	odp_schedule_group_t (*queue_sched_group)(odp_queue_t queue);
+	int (*queue_lock_count)(odp_queue_t queue);
+	uint64_t (*queue_to_u64)(odp_queue_t hdl);
+	void (*queue_param_init)(odp_queue_param_t *param);
+	int (*queue_info)(odp_queue_t queue, odp_queue_info_t *info);
+} queue_api_t;
+
+/* Queue functions towards other internal components */
+typedef struct {
+	int (*queue_init_global)(void);
+	int (*queue_term_global)(void);
+	int (*queue_enq)(queue_entry_t *queue, odp_buffer_hdr_t *buf_hdr);
+	int (*queue_enq_multi)(queue_entry_t *queue,
+			       odp_buffer_hdr_t *buf_hdr[], int num);
+	odp_buffer_hdr_t *(*queue_deq)(queue_entry_t *queue);
+	int (*queue_deq_multi)(queue_entry_t *queue,
+			       odp_buffer_hdr_t *buf_hdr[], int num);
+} queue_fn_t;
+
+const queue_fn_t *queue_fn;
```

Comment:
odp_queue_internal.h will be internal to default queue implementation. odp_scheduler_if.h will be the external file that will be included by all other components.